### PR TITLE
Make literals an error for tight packing (experimental 0.5.0)

### LINKED ISSUE #982: Tight packing should take number constants as uint256
Unless explicitly typecasted to a shorter type.

I'm not fully sure which is the better behaviour, but I think this is more explicit and less prone to error.

### PR DESCRIPTION
**Pull request by axic**
Fixes #982.

### CODE REVIEW

File: test/libsolidity/syntaxTests/tight_packing_literals_050.sol [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    }
+    function k() pure public returns (bytes) {
+        return abi.encodePacked(1);
+    }
+}
+
+// ----
+// TypeError: (117-118): The type of "int_const 1" was inferred as uint8. This is probably not desired. Use an explicit type.
+// Warning: (191-198): "sha3" has been deprecated in favour of "keccak256"
+// TypeError: (196-197): The type of "int_const 1" was inferred as uint8. This is probably not desired. Use an explicit type.
```
Comment (chriseth) [inline test/libsolidity/syntaxTests/tight_packing_literals_050.sol]: The message sounds a little weird, especially with these short sentences. What about `Cannot perform packed encoding for a literal. Please convert it to an explicit type first.`

**Review by chriseth — DISMISSED** [review]

**Comment by axic** [comment]
@chriseth updated

**Review by chriseth — APPROVED** [review]

**Review by chriseth — APPROVED** [review]
We are currently only handling rational number types. I guess the issue is easier for strings, but we should check.

**Comment by chriseth** [comment]
also things like `hex"abcd"`

**Comment by axic** [comment]
I'd merge this as is and create a new PR which, if needed, fixes all these for string literals.

**Comment by axic** [comment]
@chriseth created an issue to track it: #3962

**Pull request merged**